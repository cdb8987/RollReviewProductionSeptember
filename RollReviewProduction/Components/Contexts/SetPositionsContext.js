import {createContext} from 'react'
export const SetPositionsContext = createContext()

