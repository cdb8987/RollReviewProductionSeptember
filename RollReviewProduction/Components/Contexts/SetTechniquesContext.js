import {createContext} from 'react'
export const SetTechniquesContext = createContext()

