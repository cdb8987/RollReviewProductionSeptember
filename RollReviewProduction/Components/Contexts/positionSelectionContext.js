import {createContext} from 'react'
export const positionSelectionContext = createContext()