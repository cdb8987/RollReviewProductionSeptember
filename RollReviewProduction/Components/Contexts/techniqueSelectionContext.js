import {createContext} from 'react'
export const techniqueSelectionContext = createContext()