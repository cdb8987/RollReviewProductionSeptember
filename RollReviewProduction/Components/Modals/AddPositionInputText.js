
import React from 'react'
import {useState} from 'react'
import { Button, View}from 'react-native'
import {TextInput} from 'react-native-paper'


export default function AddPositionInputText({handleAdd,positions, setPositions, onDismiss }){

        const [text, setText] = useState('')
        

        return (
            <>
            <TextInput
                style={{height: 40, flex: 1, backgroundColor: 'white'}}
                placeholder="..."
                onChangeText={(newText) => {setText(newText)}}
                defaultValue={text}
                outlined={true}
                />
            <Button 
                style={{marginTop: 15}}
                compact={true} 
                mode={'contained'} 
                onPress={()=>{
                    const newItem = text.charAt(0).toUpperCase()+text.slice(1)
                    if(newItem && !positions.includes(newItem)){
                    handleAdd(positions, setPositions, newItem)
                    onDismiss()}
                    
                    else{ onDismiss()}
                }}
                title={'ADD'}
                
            ></Button>
            </>

        )

        
}



