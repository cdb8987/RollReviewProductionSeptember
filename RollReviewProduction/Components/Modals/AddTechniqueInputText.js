import React from 'react'
import {useState} from 'react'
import { Button}from 'react-native'
import {TextInput} from 'react-native-paper'


export default function AddTechniqueInputText({handleAdd,techniques, setTechniques, onDismiss}){
        const [text, setText] = useState('')

        return (
            <>
            <TextInput
                style={{height: 40, flex: 1, backgroundColor: 'white'}}
                placeholder="..."
                onChangeText={(newText) => {setText(newText)}}
                defaultValue={text}
                outlined={true}
                />
            <Button 
                style={{marginTop: 15}}
                compact={true} 
                mode={'contained'} 
                onPress={()=>{
                    const newItem = text.charAt(0).toUpperCase()+text.slice(1)
                    if(newItem && !techniques.includes(newItem)){console.log(text,  'added')
                    handleAdd(techniques, setTechniques, newItem)
                    onDismiss()}
                    
                    else{console.log(newItem, 'already in list'); onDismiss()}
                }}
                title={'ADD'}
                
            ></Button>
            </>

        )
}



