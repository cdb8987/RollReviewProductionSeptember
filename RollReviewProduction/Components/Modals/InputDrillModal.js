import AsyncStorage from '@react-native-async-storage/async-storage';
import { StyleSheet, Text, View } from 'react-native';
import React from "react";
import Notesinput from './NotesInput';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { useContext, useState, useRef } from 'react'
import { PositionSelectionChoicesContext } from '../Contexts/PositionsSelectionChoicesContext';
import { TechniqueSelectionChoicesContext } from '../Contexts/TechniqueSelectionChoicesContext';
import { Modal, Portal, Button} from 'react-native-paper';
import { Appbar } from 'react-native-paper';
import { techniqueSelectionContext } from '../Contexts/techniqueSelectionContext';
import { positionSelectionContext } from '../Contexts/positionSelectionContext';
import { SetPositionsContext } from '../Contexts/SetPositionsContext';
import { SetTechniquesContext } from '../Contexts/SetTechniquesContext';
import DropdownSingleSelect from '../DropdownSingleSelect';
import {Foundation} from '@expo/vector-icons'
import AddInputPositionText from './AddPositionInputText';
import AddTechniqueInputText from './AddTechniqueInputText'
import { handleAddPosition, handleAddTechnique } from '../../Functions/functions';


export default function InputDrillModal({
  visible, 
  setVisible
  
}){
    
    const showModal = () => setVisible(true);
    const hideModal = () => setVisible(false);
    const containerStyle = {backgroundColor: 'white', padding: 20, flex: 1};
    // const SelectionChoices = useContext(SelectionChoicesContext)
    const positions = useContext(PositionSelectionChoicesContext)
    const techniques = useContext(TechniqueSelectionChoicesContext)
    const setPositions = useContext(SetPositionsContext)
    const setTechniques = useContext(SetTechniquesContext)
    const {positionSelection, setpositionSelection} = useContext(positionSelectionContext)
    const {techniqueSelection, setTechniqueSelection} = useContext(techniqueSelectionContext)
    const [addPositionModalVisible, setAddPositionModalVisible] = useState(false)
    const [addTechniqueModalVisible, setAddTechniqueModalVisible] = useState(false)
    
    
    
    
    const TopBar = () => (
        <Appbar.Header>
          <Appbar.BackAction onPress={() => {hideModal()}} />
          <Text>DRILL</Text>
        </Appbar.Header>
    );


    const AddPositionTextInputModal = ()=>{
      return (
        <Modal style={{flex: 1}} transparent={true}visible={addPositionModalVisible} onDismiss={hideAddPositionModal} contentContainerStyle={containerStyle}>
            <Appbar.Header>
              <Appbar.BackAction onPress={() => {hideAddPositionModal()}} />
                <Text>POSITION</Text>
                <AddInputPositionText 
                  handleAdd={handleAddPosition}
                  positions={positions}
                  setPositions={setPositions}
                  onDismiss={hideAddPositionModal}
                  />
                
            </Appbar.Header>
            
            
            </Modal>
          
      )
    }

    const AddTechniqueTextInputModal = ()=>{


      
      return (
        <Modal style={{flex: 1}} transparent={true}visible={addTechniqueModalVisible} onDismiss={hideAddTechniqueModal} contentContainerStyle={containerStyle}>
            <Appbar.Header>
              <Appbar.BackAction onPress={() => {hideAddTechniqueModal()}} />
                <Text>TECHNIQUE</Text>
                <AddTechniqueInputText 
                  handleAdd={handleAddTechnique}
                  techniques={techniques}
                  setTechniques={setTechniques}
                  onDismiss={hideAddTechniqueModal}
                  />
                
            </Appbar.Header>
            
            
            </Modal>
          
      )
    }



      const hideAddPositionModal = ()=>setAddPositionModalVisible(false)
      const hideAddTechniqueModal = ()=>setAddTechniqueModalVisible(false)




    let drillRecord = useRef({});

    const updateDrillPositionSelection = function(positionChoice){
      drillRecord['position'] = positionChoice;
     
    }
    const updateDrillTechniqueSelection = function(techniqueChoice){
      drillRecord['technique'] = techniqueChoice
    }
    const updateDrillRoundsSelection = function(roundsChoice){
      drillRecord['rounds'] = roundsChoice
    }
    const updateDrillNotes = (notes)=>{
      drillRecord['notes'] = notes;
    }
    const handleLogDrill = async () => {
      let drillingHistory;
      
      const getData = async ()=>{
        if(await AsyncStorage.getItem('drillingHistory')===null){
          const initialData = await JSON.stringify({ "data":[] })
          await AsyncStorage.setItem('drillingHistory', initialData)
        }
        let storedData = await AsyncStorage.getItem('drillingHistory')
        return storedData
      }      
      const updateData = async (storedData)=>{
        if(!drillRecord.rounds){console.log('no rounds!');return null}
        drillingHistory = await JSON.parse(storedData)
        const d = new Date()
        drillRecord['createdAt'] = d
        drillRecord['recordType'] = 'drill'
        drillRecord['recordID'] = d.getTime()
        for(let i of ['rounds', 'position', 'technique', 'notes' ]){
          if(!drillRecord[i]){
            drillRecord[i]=''
          }
        }

        drillingHistory.data.push(drillRecord)
        await AsyncStorage.setItem('drillingHistory', JSON.stringify(drillingHistory))
        
      }
      
      await updateData(await getData())

      setpositionSelection('placeholder') //triggers global re-render and persists values
      setpositionSelection(String(positionSelection))
      setTechniqueSelection('placeholder')
      setTechniqueSelection(String(techniqueSelection))
    
    

    }

    return(
        
        <Portal>
        <Modal style={{flex: 1}} visible={visible} onDismiss={hideModal} contentContainerStyle={containerStyle}>
          
        <KeyboardAwareScrollView>
        <View style={styles.containerStyle} >
        
            <TopBar/>
            <View style={styles.TextInputContainer}>
                
                <View style={{flexDirection: 'row'}}>
                <View style={{flex: 5}}>
                  <DropdownSingleSelect 
                  inputOptions={positions} 
                  labelName={'Position'} 
                  onSelectCallback={updateDrillPositionSelection}/>
                </View>
                <View style={{flex: 1}}>
                  <Button onPress={()=>{setAddPositionModalVisible(true)}}><Foundation name="plus" size={24} color="black" /></Button>
                </View>
                </View>



                <View style={{flexDirection: 'row'}}>
                <View style={{flex: 5}}>
                  <DropdownSingleSelect 
                  inputOptions={techniques} 
                  labelName={'Technique'} 
                  onSelectCallback={updateDrillTechniqueSelection}/>
                </View>
                <View style={{flex: 1}}>
                  <Button onPress={()=>setAddTechniqueModalVisible(true)}><Foundation name="plus" size={24} color="black" /></Button>
                </View>
                </View>

 
                <DropdownSingleSelect
                style={styles.dropdown} 
                inputOptions={['1', '2', '3', '4', '5', '6', '7', '8', '9']} 
                labelName={'Rounds'} 
                onSelectCallback={updateDrillRoundsSelection}/>
                

            </View>
            <View style={styles.DropdownContainerverticalpadding}>
            </View>
            
            
            <View style={styles.notes}>
              <Notesinput updateNotes={updateDrillNotes}/>
            </View>
            
            
            
            
            
            <Button style={styles.button} 
               
              buttonColor="white"
              textColor='black'
              onPress={(async () => {
                handleLogDrill('test'); 
                hideModal()
                
              })}
              mode="outlined"
            >
             LOG DRILL </Button>
              <View style={styles.DropdownContainerverticalpadding}></View>      
        </View>
        </KeyboardAwareScrollView>
        </Modal>
         <AddPositionTextInputModal />
         <AddTechniqueTextInputModal/> 
        </Portal>

    )

}

const styles = StyleSheet.create({
    containerStyle: {
      width: "100%",
      flex: 1,

      backgroundColor: 'white',
      
     
    },
    
    button:{
      flex: 2,
      width: "100%",
      alignItems: 'center',
      justifyContent: 'center', 
    backgroundColor: '#6699ff'
  }, 
  TextInputContainer:{
      backgroundColor: 'white',
      flex: 2,
  },
    DropdownContainerverticalpadding: {
      flex: 1
  },
  header:{
    backgroundColor: 'white',
    fontSize: 25,
    fontWeight: 'bold', 

  }, 
  image: {
    flex: 1,
    justifyContent: 'center',
  },
  notes:{
    flex: 1,
    backgroundColor: 'white',
  }, 
  dropdown:{
    flex: 7

  }, 
  addButton:{
    flex: 1
  }
  
  });