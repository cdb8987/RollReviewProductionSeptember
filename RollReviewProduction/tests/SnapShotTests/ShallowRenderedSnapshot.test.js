import React from "react";
import {act} from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import AddPositionInputText from '../../Components/Modals/AddPositionInputText.js'
import AddTechniqueInputText from '../../Components/Modals/AddTechniqueInputText.js'
// import EventTaggingModal from '../../Components/Modals/EventTaggingModal.js'
import InputDrillModal from '../../Components/Modals/InputDrillModal.js'
// import InputLocalVideoModal from '../../Components/Modals/InputLocalVideoModal.js'
// import InputRollModal from '../../Components/Modals/InputRollModal.js'
// import InputTutorialModal from '../../Components/Modals/InputTutorialModal.js'
// import ModalContainer from '../../Components/Modals/ModalContainer.js'
// import Notesinput from '../../Components/Modals/Notesinput.js'  


import { Button}from 'react-native'
import {TextInput} from 'react-native-paper'
import { SetPositionsContext } from "../../Components/Contexts/SetPositionsContext.js";
import { SetTechniquesContext } from "../../Components/Contexts/SetTechniquesContext.js";
import { techniqueSelectionContext } from "../../Components/Contexts/techniqueSelectionContext.js";
import { positionSelectionContext } from "../../Components/Contexts/positionSelectionContext.js";



jest.mock('./SetPositionsContext');
jest.mock('./SetTechniquesContext');
jest.mock('./techniqueSelectionContext');
jest.mock('./positionSelectionContext');




// //Checks Shallow Rendered Snapshot for AddPositionInputText.js
// test('Shallow Render matches snapshot: AddPositionInputText',  () => {
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<AddPositionInputText 
//     />);
    
//     expect(tree).toMatchSnapshot();
    
// });

// //Checks Shallow Rendered Snapshot for AddTechniqueInputText.js
// test('Shallow Render matches snapshot: AddTechniqueInputText',  () => {
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<AddTechniqueInputText 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });

// //Checks Shallow Rendered Snapshot for EnterURLInputbox.js
// test('Shallow Render matches snapshot: EnterURLInputbox',  () => {
    
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<EventTaggingModal 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });

//Checks Shallow Rendered Snapshot for InputDrillModal.js


test('Shallow Render matches snapshot: InputDrillModal',  () => {
    
    SetPositionsContext.mockReturnValue({
    setPositions: jest.fn(), 
  });
    SetTechniquesContext.mockReturnValue({
    SetTechniquesContext: jest.fn(), 
  });
    techniqueSelectionContext.mockReturnValue({
    techniqueSelectionContext: '', 
    });

    positionSelectionContext.mockReturnValue({
    positionSelectionContext: '', 
  });





    const renderer = new ShallowRenderer()
    const tree = renderer.render(<InputDrillModal 
    />);
    

    expect(tree).toMatchSnapshot();
    
});

// //Checks Shallow Rendered Snapshot for InputLocalVideoModal.js
// test('Shallow Render matches snapshot: InputLocalVideoModal',  () => {
    
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<InputLocalVideoModal 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });

// //Checks Shallow Rendered Snapshot for InputRollModal.js
// test('Shallow Render matches snapshot: InputRollModal',  () => {
    
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<InputRollModal 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });

// //Checks Shallow Rendered Snapshot for InputTutorialModal.js
// test('Shallow Render matches snapshot: InputTutorialModal',  () => {
    
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<InputTutorialModal 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });

// //Checks Shallow Rendered Snapshot for ModalContainer.js
// test('Shallow Render matches snapshot: ModalContainer',  () => {
    
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<ModalContainer 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });

// //Checks Shallow Rendered Snapshot for Notesinput.js
// test('Shallow Render matches snapshot: Notesinput',  () => {
    
//     const renderer = new ShallowRenderer()
//     const tree = renderer.render(<Notesinput 
//     />);
    

//     expect(tree).toMatchSnapshot();
    
// });