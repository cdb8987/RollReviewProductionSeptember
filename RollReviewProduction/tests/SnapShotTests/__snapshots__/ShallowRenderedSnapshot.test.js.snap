// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render matches snapshot: AddPositionInputText 1`] = `
<React.Fragment>
  <ForwardRef
    defaultValue=""
    onChangeText={[Function]}
    outlined={true}
    placeholder="..."
    style={
      {
        "backgroundColor": "white",
        "flex": 1,
        "height": 40,
      }
    }
  />
  <Button
    compact={true}
    mode="contained"
    onPress={[Function]}
    style={
      {
        "marginTop": 15,
      }
    }
    title="ADD"
  />
</React.Fragment>
`;

exports[`Shallow Render matches snapshot: AddTechniqueInputText 1`] = `
<React.Fragment>
  <ForwardRef
    defaultValue=""
    onChangeText={[Function]}
    outlined={true}
    placeholder="..."
    style={
      {
        "backgroundColor": "white",
        "flex": 1,
        "height": 40,
      }
    }
  />
  <Button
    compact={true}
    mode="contained"
    onPress={[Function]}
    style={
      {
        "marginTop": 15,
      }
    }
    title="ADD"
  />
</React.Fragment>
`;

exports[`Shallow Render matches snapshot: EnterURLInputbox 1`] = `
<withTheme(Portal)>
  <Modal
    contentContainerStyle={
      {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 20,
      }
    }
    onDismiss={[Function]}
    style={
      {
        "flex": 1,
      }
    }
  >
    <KeyboardAwareScrollView
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardOpeningTime={250}
      viewIsInsideTabBar={false}
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "flex": 1,
            "width": "100%",
          }
        }
      >
        <TopBar />
        <View
          style={
            {
              "backgroundColor": "white",
              "flex": 2,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "flex": 5,
                }
              }
            >
              <DropdownSingleSelect
                labelName="Position"
                onSelectCallback={[Function]}
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <ForwardRef(Button)
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="black"
                  name="plus"
                  size={24}
                />
              </ForwardRef(Button)>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "flex": 5,
                }
              }
            >
              <DropdownSingleSelect
                labelName="Technique"
                onSelectCallback={[Function]}
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <ForwardRef(Button)
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="black"
                  name="plus"
                  size={24}
                />
              </ForwardRef(Button)>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "white",
              "flex": 1,
            }
          }
        >
          <Notesinput
            updateNotes={[Function]}
          />
        </View>
        <ForwardRef(Button)
          buttonColor="white"
          mode="outlined"
          onPress={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#6699ff",
              "flex": 2,
              "justifyContent": "center",
              "width": "100%",
            }
          }
          textColor="black"
        >
          TAG 
           
        </ForwardRef(Button)>
        <View
          style={
            {
              "flex": 1,
            }
          }
        />
      </View>
    </KeyboardAwareScrollView>
  </Modal>
  <AddPositionTextInputModal />
  <AddTechniqueTextInputModal />
</withTheme(Portal)>
`;

exports[`Shallow Render matches snapshot: ModalContainer 1`] = `
<React.Fragment>
  <InputDrillModal
    setVisible={[Function]}
    visible={false}
  />
  <InputRollModal
    setVisible={[Function]}
    visible={false}
  />
  <InputLocalVideoModal
    setVisible={[Function]}
    visible={false}
  />
  <InputTutorialModal
    setVisible={[Function]}
    visible={false}
  />
  <RedPlusFAB
    setInputDrillModalVisible={[Function]}
    setInputRollModalVisible={[Function]}
    setInputTutorialModalVisible={[Function]}
    setLocalVideoModalVisible={[Function]}
    style={
      {
        "marginBottom": 110,
        "marginRight": 10,
        "position": "absolute",
      }
    }
    visible={true}
  />
</React.Fragment>
`;
